client/src/components/Attachment/UploadAttachment.js

```diff
@@ -25,26 +25,52 @@ const GQL_UPDATE_ATTACHMENT = gql`
   }
 `
 
-const UploadAttachment = ({ getRelatedObject, edit, saveRelatedObject }) => {
+const UploadAttachment = ({
+  edit,
+  saveRelatedObject,
+  relatedObjectType,
+  relatedObjectUuid
+}) => {
   const [error, setError] = useState(null)
   const [remove, setRemove] = useState(false)
   const [uploadedList, setUploadedList] = useState([])
-  const relatedObject = getRelatedObject()
 
-  const attachmentSave = async e => {
+  useEffect(() => {
+    // In the edit mode the base page entity (report, location, etc.) always has an uuid
+    // So when uploading new attachment it is always inserted related tables, (ie. see attachmentSave)
+    if (edit) {
+      return
+    }
+    // if base entity is not created, the attachment can not be associated
+    if (!relatedObjectUuid) {
+      return
+    }
+    // Update only object not associated with the base entity, as in report
+    const updateList = uploadedList.filter(
+      list => !list.attachmentRelatedObjects[0].relatedObjectUuid
+    )
+    for (let i = 0; i < updateList.length; i++) {
+      save(updateList[i], relatedObjectUuid, true)
+    }
+  }, [edit, relatedObjectUuid, uploadedList])
+
+  const attachmentSave = async(e, relatedUuid) => {
+    if (!e.target.files[0]) {
```

Comment:
You could change code like
`const file = e.target.files[0]
if(!file) {return}`